source/unix/opensslcrypto_hash.c

```diff
@@ -99,6 +107,35 @@ struct aws_hash *aws_sha256_default_new(struct aws_allocator *allocator) {
     return hash;
 }
 
+struct aws_hash *aws_sha1_default_new(struct aws_allocator *allocator) {
+    struct aws_hash *hash = aws_mem_acquire(allocator, sizeof(struct aws_hash));
+
+    if (!hash) {
+        return NULL;
+    }
+
+    hash->allocator = allocator;
+    hash->vtable = &s_sha1_vtable;
+    hash->digest_size = AWS_SHA1_LEN;
+    EVP_MD_CTX *ctx = g_aws_openssl_evp_md_ctx_table->new_fn();
+    hash->impl = ctx;
+    hash->good = true;
+
+    if (!hash->impl) {
+        s_destroy(hash);
+        aws_raise_error(AWS_ERROR_OOM);
+        return NULL;
+    }
+
+    if (!g_aws_openssl_evp_md_ctx_table->init_ex_fn(ctx, EVP_sha1(), NULL)) {
+        s_destroy(hash);
```

Comment:
we need to clean up whatever was created by `EVP_MD_CTX_new()`.
I can't focus enough to figure out which function pointer pointer pointer that is

fix this in the other implementations as well

Reply:
nevermind. the s_destroy() function handles it as you've pointed out